File: ruby/engine/test/ReportingMeasureWithModelOutputs/measure.rb
# insert your copyright here

# see the URL below for information on how to write OpenStudio measures
# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/

require 'erb'

# start the measure
class ReportingMeasureWithModelOutputs < OpenStudio::Measure::ReportingMeasure
  # human readable name
  def name
    # Measure name should be the title case of the class name.
    return 'ReportingMeasureWithModelOutputs'
  end

  # human readable description
  def description
    return 'DESCRIPTION_TEXT'
  end

  # human readable description of modeling approach
  def modeler_description
    return 'MODELER_DESCRIPTION_TEXT'
  end

  # define the arguments that the user will input
  def arguments(model = nil)
    args = OpenStudio::Measure::OSArgumentVector.new

    return args
  end

  # define the outputs that the measure will create
  def outputs
    outs = OpenStudio::Measure::OSOutputVector.new

    # this measure does not produce machine readable outputs with registerValue, return an empty list

    return outs
  end

  def modelOutputRequests(model, runner, user_arguments)
    return true
  end

  # return a vector of IdfObject's to request EnergyPlus objects needed by the run method
  # Warning: Do not change the name of this method to be snake_case. The method must be lowerCamelCase.
  def energyPlusOutputRequests(runner, user_arguments)
    result = OpenStudio::IdfObjectVector.new

    return result
  end

  # define what happens when the measure is run
  def run(runner, user_arguments)
    super(runner, user_arguments)

    return true
  end
end

# register the measure to be used by the application
ReportingMeasureWithModelOutputs.new.registerWithApplication

File: python/engine/test/ReportingMeasureWithoutModelOutputs/measure.py
"""insert your copyright here.

# see the URL below for information on how to write OpenStudio measures
# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/
"""

from pathlib import Path

import openstudio


class ReportingMeasureWithoutModelOutputs(openstudio.measure.ReportingMeasure):
    """An ReportingMeasure."""

    def name(self):
        return "ReportingMeasureWithoutModelOutputs"

    def description(self):
        return "DESCRIPTION_TEXT"

    def modeler_description(self):
        return "MODELER_DESCRIPTION_TEXT"

    def arguments(self, model: openstudio.model.Model):
        args = openstudio.measure.OSArgumentVector()
        return args

    def outputs(self):
        outs = openstudio.measure.OSOutputVector()
        return outs

    def energyPlusOutputRequests(
        self, runner: openstudio.measure.OSRunner, user_arguments: openstudio.measure.OSArgumentMap
    ):
        result = openstudio.IdfObjectVector()
        return result

    def run(
        self,
        runner: openstudio.measure.OSRunner,
        user_arguments: openstudio.measure.OSArgumentMap,
    ):
        super().run(runner, user_arguments)
        return True
